lib/puppet/pops/types/type_calculator.rb

```diff
@@ -644,6 +644,11 @@ def infer_Symbol(o)
   end
 
   # @api private
+  def infer_Sensitive(o)
+    PSensitiveType::DEFAULT
```

Comment:
This would make all sensitive instances be of the type `Sensitive[Any]`. It should instead infer the type of o and pass that on as a parameter. That is:
```
PSensitiveType.new(infer(o.value))
```